CRM/MembersOnlyEvent/Form/MembersOnlyEventTab.php

```diff
@@ -314,6 +330,12 @@ private function saveFormData($params) {
       }
 
       EventMembershipType::updateAllowedMembershipTypes($membersOnlyEvent->id, $allowedMembershipTypesIDs);
+
+      $nonMemberPriceFieldValueIDs = array();
+      if (!empty($params['non_member_price_field_values'])) {
+        $nonMemberPriceFieldValueIDs = $params['non_member_price_field_values'];
+      }
+      MembersOnlyEventPriceFieldValue::updateNonMemberPriceFieldValues($membersOnlyEvent->id, $nonMemberPriceFieldValueIDs);
```

Comment:
Perhaps, it is better to have a logic to check in BAO if the saved values have been updated, and we should only update when the field has been updated?

Reply:
Yeah, I've add another commit for that